apps/ledger-live-mobile/src/components/DeviceAction/rendering.tsx

```diff
@@ -1190,3 +1191,251 @@ export const renderImageCommitRequested = ({
+      <Text fontSize={24} fontWeight="semiBold" textAlign="center" mt={6}>
+        {t("FirmwareUpdate.updateCancelled", {
+          deviceName: getDeviceModel(device.modelId).productName,
+        })}
+      </Text>
+      <Text fontSize={14} textAlign="center" color="neutral.c80" mt={6}>
+        {t("FirmwareUpdate.updateCancelledDescription", {
+          deviceName: getDeviceModel(device.modelId).productName,
+          newFirmwareVersion,
+        })}
+      </Text>
+      <Button
+        type="main"
+        outline={false}
+        onPress={onPressRestart}
+        mt={8}
+        alignSelf="stretch"
+      >
+        {t("FirmwareUpdate.restartUpdate")}
+      </Button>
+      <Button
+        type="default"
+        outline={false}
+        onPress={onPressQuit}
+        mt={6}
+        alignSelf="stretch"
+      >
+        {t("FirmwareUpdate.quitUpdate")}
+      </Button>
+    </Flex>
+  );
+};
+
+export const FirmwareUpdateError = ({
+  t,
+  device,
+  onPressQuit,
+  errorName,
+}: {
+  t: TFunction;
+  device: Device;
+  onPressQuit: () => void;
+  errorName: string;
+}) => {
+  const { errorTitle, errorDescription } = useMemo(() => {
+    const firmwareUpdateErrorTitleId = `FirmwareUpdate.errors.${errorName}.title`;
+    const firmwareUpdateErrorDescriptionId = `FirmwareUpdate.errors.${errorName}.description`;
+
+    const genericErrorTitleId = `errors.${errorName}.title`;
+    const genericErrorDescriptionId = `errors.${errorName}.description`;
+
+    const firmwareUpdateErrorTitle = t(firmwareUpdateErrorTitleId, {
+      deviceName: getDeviceModel(device.modelId).productName,
+    });
+    const firmwareUpdateErrorDescription = t(firmwareUpdateErrorDescriptionId, {
+      deviceName: getDeviceModel(device.modelId).productName,
+    });
+
+    const genericErrorTitle = t(genericErrorTitleId, {
+      deviceName: getDeviceModel(device.modelId).productName,
+    });
+    const genericErrorDescription = t(genericErrorDescriptionId, {
+      deviceName: getDeviceModel(device.modelId).productName,
+    });
+
+    // this function returns the translated strings for the error's description and title
+    // it gives priority to the translations that exists in the context of the firmware update
+    // if there is no specific translation for the context of the firmware update it fetches
+    // the generic error translations
+    // it also considers that there might not be a translated description for the error, in which case
+    // it returns "null" for the description
+    return {
+      errorTitle:
+        firmwareUpdateErrorTitle !== firmwareUpdateErrorTitleId
+          ? firmwareUpdateErrorTitle
+          : genericErrorTitle,
+      errorDescription:
+        firmwareUpdateErrorDescription !== firmwareUpdateErrorDescriptionId
+          ? firmwareUpdateErrorDescription
+          : genericErrorDescription !== genericErrorDescriptionId
+          ? genericErrorDescription
+          : null,
+    };
+  }, [errorName, t]);
```

Comment:
I've made this generic by adding an optional context. I'm not sure this is the same logic that exists in the `TranslatedError` component though, it might make it even harder to understand the logic of `TranslatedError` to accept my use case.